layers/core_checks/cc_data_graph.cpp

```diff
@@ -0,0 +1,429 @@
+                    "(%" PRIu32 ") must be 0", pipeline_feedback->pipelineStageCreationFeedbackCount);
+            }
+        }
+        auto layout_ptr = Get<vvl::PipelineLayout>(create_info.layout);
+        ASSERT_AND_RETURN_SKIP(layout_ptr);
+        if (layout_ptr->push_constant_ranges_layout->size() != 0) {
+            skip |= LogError("VUID-VkDataGraphPipelineCreateInfoARM-layout-09767", device,
+                             create_info_loc.dot(Field::layout).dot(Field::pushConstantRangeCount), "(%zu) must be 0",
+                             layout_ptr->push_constant_ranges_layout->size());
+        }
+        for (size_t j = 0; j < layout_ptr->set_layouts.size(); j++) {
+            auto dsl = layout_ptr->set_layouts[j];
+            if ((dsl->GetCreateFlags() & VK_DESCRIPTOR_SET_LAYOUT_CREATE_UPDATE_AFTER_BIND_POOL_BIT) != 0) {
+                if (!enabled_features.dataGraphUpdateAfterBind) {
+                    skip |= LogError("VUID-VkDataGraphPipelineCreateInfoARM-dataGraphUpdateAfterBind-09768", device,
+                                        create_info_loc.dot(Field::layout),
+                                        "descriptorSetLayout created with (%s) but dataGraphUpdateAfterBind not enabled",
+                                        string_VkDescriptorSetLayoutCreateFlags(dsl->GetCreateFlags()).c_str());
+                }
+            }
+            auto bindings = dsl->GetBindings();
+            for (auto binding : bindings) {
+                auto mutable_bindings = dsl->GetMutableTypes(binding.binding);
+                if (mutable_bindings.size() != 0) {
+                    skip |= LogError("VUID-VkDataGraphPipelineCreateInfoARM-pSetLayouts-09770", device,
+                                        create_info_loc.dot(Field::pSetLayouts),
+                                        "Cannot use a VK_DESCRIPTOR_TYPE_MUTABLE_EXT in a DataGraphPipeline");
+                }
+            }
+        }
+        if ((create_info.flags & (VK_PIPELINE_CREATE_2_FAIL_ON_PIPELINE_COMPILE_REQUIRED_BIT_KHR |
+                                    VK_PIPELINE_CREATE_2_EARLY_RETURN_ON_FAILURE_BIT_KHR)) != 0) {
+            if (!enabled_features.pipelineCreationCacheControl) {
+                skip |= LogError(
+                    "VUID-VkDataGraphPipelineCreateInfoARM-pipelineCreationCacheControl-09871", device, create_info_loc.dot(Field::flags),
+                    "is (%s) but pipelineCreationCacheControl is not enabled",
+                    string_VkPipelineCreateFlags2(create_info.flags).c_str());
+            }
+        }
+
+        for (uint32_t j = 0; j < create_info.resourceInfoCount; j++) {
+            auto resource = create_info.pResourceInfos[j];
+            if (resource.arrayElement != 0) {
+                skip |= LogError("VUID-VkDataGraphPipelineResourceInfoARM-arrayElement-09779", device,
+                                 create_info_loc.dot(Field::arrayElement), "(%" PRIu32 ") not 0", resource.arrayElement);
```

Comment:
```suggestion
                                 create_info_loc.dot(Field::arrayElement), "(%" PRIu32 ") is not zero", resource.arrayElement);
```

is our wording for these